app/controllers/pages_controller.rb

```diff
@@ -136,4 +137,8 @@ def not_found_conditions
     not_found unless FeatureFlag.accessible?(@page.feature_flag_name, current_user)
     not_found if params[:format] == "txt" && @page.template != "txt"
   end
+
+  def combined_fragmented_slug
+    (0..5).filter_map { |i| params["slug_#{i}"] }.join("/")
```

Comment:
nit: Might be a good idea to reference the validator's variable here instead.
```ruby
(0...CrossModelSlugValidator::PAGE_DIRECTORY_LIMIT).filter_map { ... }
```

Reply:
Good call. I'm going to include that in a follow-up